backend/src/main/java/eu/clarin/switchboard/resources/UrlMatchResource.java

```diff
@@ -0,0 +1,80 @@
+package eu.clarin.switchboard.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.clarin.switchboard.app.FileAsset;
+import eu.clarin.switchboard.app.config.ToolConfig;
+import eu.clarin.switchboard.core.MediaLibrary;
+import eu.clarin.switchboard.core.ToolRegistry;
+import eu.clarin.switchboard.core.xc.CommonException;
+import eu.clarin.switchboard.profiler.api.Profile;
+import eu.clarin.switchboard.profiler.api.ProfilingException;
+import eu.clarin.switchboard.tool.Tool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Path("/api/urlmatch")
+public class UrlMatchResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UrlMatchResource.class);
+
+    MediaLibrary mediaLibrary;
+    ToolRegistry toolRegistry;
+    ToolConfig toolConfig;
+
+    public UrlMatchResource(MediaLibrary mediaLibrary, ToolRegistry toolRegistry, ToolConfig toolConfig) {
+        this.mediaLibrary = mediaLibrary;
+        this.toolRegistry = toolRegistry;
+        this.toolConfig = toolConfig;
+    }
+
+    /**
+     * PUBLIC API
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
+    public Response doUrlsMatch(@QueryParam("onlyProductionTools") String onlyProductionTools, List<String> urlList) throws CommonException, ProfilingException {
+        long startTime = System.nanoTime();
+        List<Profile> profiles = urlList.stream().map(url -> {
+            try {
+                return mediaLibrary.addByUrlPreflight(url, null).getProfile().toProfile();
+            } catch (Exception e) {
+                LOGGER.info("Exception in urlmatch addByUrlPreflight: "+e.getMessage());
```

Comment:
any reason this is not at warning level?

Reply:
An exception will be triggered for bad urls, for example, which can be added by any user. I want to display the exceptions because they could be program errors, but not to give them too much importance since most of them will just be HTTP failures.